src/openssl/ssl/context.cr

```diff
@@ -1,22 +1,40 @@
 class OpenSSL::SSL::Context
-  def self.default : self
-    @@default ||= new
-  end
-
   def initialize
     @handle = LibSSL.ssl_ctx_new(LibSSL.sslv23_method)
+    self.verify_mode = OpenSSL::SSL::VerifyMode::PEER
```

Comment:
@ysbaddaden did you understand what this does in server mode? In particular I wonder if setting this in server mode will cause prompts in Browsers etc.

> SSL_VERIFY_PEER
           Server mode: the server sends a client certificate request to the client.  The certificate returned (if any) is checked. If the verification process
           fails, the TLS/SSL handshake is immediately terminated with an alert message containing the reason for the verification failure.